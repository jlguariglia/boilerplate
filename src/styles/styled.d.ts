import 'styled-components';

interface Colors {
  main: string;
  contrastText: string;
}

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      common: {
        black: string;
        white: string;
      };
      primary: Colors;
      secondary: Colors;
    };
    font: {
      family: Array;
      size: {
        small: string;
        medium: string;
        large: string;
      };
    };
    breakpoint: {
      sm: string;
      md: string;
      lg: string;
      xl: string;
    };
  }
}
