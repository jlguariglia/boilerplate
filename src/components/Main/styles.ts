import styled from 'styled-components';

export const Container = styled.main`
  background-color: ${({ theme }) => theme.colors.primary.main};
  color: ${({ theme }) => theme.colors.primary.contrastText};
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: flex;
  text-align: center;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  overflow: hidden;
`;

export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`;

export const Title = styled.h1`
  font-size: ${({ theme }) => theme.font.size.large};
  color: ${({ theme }) => theme.colors.primary.contrastText};
`;

export const Description = styled.h2`
  font-size: ${({ theme }) => theme.font.size.medium};
  font-weight: 400;
`;

export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`;
