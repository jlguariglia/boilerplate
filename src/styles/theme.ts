import { DefaultTheme } from 'styled-components';

export const defaultTheme: DefaultTheme = {
  colors: {
    common: {
      black: '#222831',
      white: '#ffffff',
    },
    primary: {
      main: '#726a95',
      contrastText: '#ffffff',
    },
    secondary: {
      main: '#709fb0',
      contrastText: '#ffffff',
    },
  },
  font: {
    size: {
      small: '1em',
      medium: '2em',
      large: '3em',
    },
    family: [
      '-apple-system',
      'BlinkMacSystemFont',
      'Segoe UI',
      'Roboto',
      'Oxygen',
      'Ubuntu',
      'Cantarell',
      'Open Sans',
      'Helvetica Neue',
      'sans-serif',
    ],
  },
  breakpoint: {
    sm: 'screen and (max-width: 640px)',
    md: 'screen and (max-width: 768px)',
    lg: 'screen and (max-width: 1024px)',
    xl: 'screen and (max-width: 1280px)',
  },
};
