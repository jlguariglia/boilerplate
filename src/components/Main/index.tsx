import { Container, Logo, Title, Description, Illustration } from './styles';

const Main = ({
  title = 'Next App - Boilerplate',
  description = 'Boilerplate utilizing TypeScript, ReactJS, NextJS and Styled Components',
}) => (
  <Container>
    <Logo
      src="/img/logo.svg"
      alt="Atom and React Avancado written right next to it"
    />
    <Title>{title}</Title>
    <Description>{description}</Description>
    <Illustration
      src="/img/hero-illustration.svg"
      alt="A software developer staring at a screen with some code on it"
    />
  </Container>
);

export default Main;
