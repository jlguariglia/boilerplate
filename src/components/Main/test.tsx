import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from 'styles/theme';

import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Main />
      </ThemeProvider>
    );

    expect(
      screen.getByRole('heading', { name: /Next App/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render colors correctly', () => {
    const { container } = render(
      <ThemeProvider theme={defaultTheme}>
        <Main />
      </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyle({ 'background-color': '#726a95' });
  });
});
