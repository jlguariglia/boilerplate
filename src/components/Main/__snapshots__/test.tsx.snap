// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #726a95;
  color: #ffffff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 3em;
  color: #ffffff;
}

.c3 {
  font-size: 2em;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Atom and React Avancado written right next to it"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    Next App - Boilerplate
  </h1>
  <h2
    class="c3"
  >
    Boilerplate utilizing TypeScript, ReactJS, NextJS and Styled Components
  </h2>
  <img
    alt="A software developer staring at a screen with some code on it"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
